include/AudioBus.h

```diff
@@ -0,0 +1,160 @@
+/*
+ * AudioBus.h
+ *
+ * Copyright (c) 2025 Dalton Messmer <messmer.dalton/at/gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LMMS_AUDIO_BUS_H
+#define LMMS_AUDIO_BUS_H
+
+#include <bitset>
+
+#include "AudioBufferView.h"
+#include "LmmsTypes.h"
+#include "lmms_constants.h"
+#include "lmms_export.h"
+#include "SampleFrame.h"
+
+namespace lmms
+{
+
+/**
+ * A non-owning span of the track channels for an instrument or effect chain
+ * which keeps track of signal flow.
+ */
+class LMMS_EXPORT AudioBus
+{
+public:
+	AudioBus() = default;
+	AudioBus(const AudioBus&) = default;
+
+	//! `bus` is assumed to be silent
+	AudioBus(SampleFrame* const* bus, track_ch_t channelPairs, f_cnt_t frames);
+
+	auto trackChannelPair(track_ch_t pairIndex) const -> InterleavedBufferView<const float, 2>
+	{
+		return {m_bus[pairIndex], m_frames};
+	}
+
+	auto trackChannelPair(track_ch_t pairIndex) -> InterleavedBufferView<float, 2>
+	{
+		return {m_bus[pairIndex], m_frames};
+	}
```

Comment:
So my idea is that we can essentially make the fact that the `AudioBus` can only work in multiples of 2 channels as an implementation detail. I think its more simple to consider `SampleFrame*` as a track channel, and then maybe add documentation explaining why it is this way. This is so that we can avoid any confusion with the terminology and possibly simplify the interface.